force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,64 @@
+trigger OpportunityTrigger on Opportunity (before update, after update, before delete) {
+
+    // Scenario 1: Validate that the amount is greater than 5000 when updated
+    if (Trigger.isUpdate) {
+        for (Opportunity opp : Trigger.new) {
+            if (opp.Amount <= 5000) {
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+
+    // Scenario 2: Prevent deletion of closed-won opportunities for banking accounts
+    if (Trigger.isDelete) {
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity opp : Trigger.old) {
+            if (opp.StageName == 'Closed Won' && opp.AccountId != null) {
+                accountIds.add(opp.AccountId);
+            }
+        }
+
+        if (!accountIds.isEmpty()) {
+            // Query accounts to check the industry
+            Map<Id, Account> accountMap = new Map<Id, Account>(
+                [SELECT Id, Industry FROM Account WHERE Id IN :accountIds]
+            );
```

Comment:
You can do a subquery here for the opportunities while getting the accounts which would avoid doing the loop on line 15